src/test/java/gov/usgs/owi/nldi/transform/CharacteristicDataTransformerTest.java

```diff
@@ -0,0 +1,66 @@
+package gov.usgs.owi.nldi.transform;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import gov.usgs.owi.nldi.dao.BaseDao;
+
+public class CharacteristicDataTransformerTest {
+
+	private static final String INIT_JSON = "{\"comid\":\"comm\",\"characteristics\":[";
+	private static final String WRITE_MAP = "{\"characteristic_id\":\"charID\",\"characteristic_value\":\"val\",\"percent_nodata\":\"none\"}";
+	protected MockHttpServletResponse response;
+	protected CharacteristicDataTransformer transformer;
+
+	@Before
+	public void beforeTest() throws IOException {
+		response = new MockHttpServletResponse();
+		transformer = new CharacteristicDataTransformer(response);
+	}
+
+	@After
+	public void afterTest() throws Exception {
+		transformer.close();
+	}
+
+
+	@Test
+	public void initJsonTest() {
+		transformer.initJson(transformer.g, buildMap());
+		try {
+			transformer.g.flush();
```

Comment:
I was a bit confused by this single-letter variable name (`g`) at first. I understand that it is short because it is convenient to reference it numerous times [elsewhere](https://github.com/USGS-CIDA/WQP-WQX-Services/blob/master/src/main/java/gov/usgs/cida/wqp/transform/MapToJsonTransformer.java#L52-L79). While it's not a blocker for this PR, in the future it might be helpful to give the short instance variables (both the generator `g` and the factory `f`) more descriptive names. The code code be kept concise via short local aliases of the longer instance variables, or by implementing a method-chaining pattern (Builder, perhaps?).